airflow/models/connection.py

```diff
@@ -186,16 +186,46 @@ def _normalize_conn_type(conn_type):
             conn_type = conn_type.replace("-", "_")
         return conn_type
 
+    @staticmethod
+    def split_uri_to_parts(uri):
+        c = uri.count("://")
+        if c < 2:
+            return urlsplit(uri), False
+        idx = uri.rfind("://")
+        # replace :// with :__
+        uri_list = list(uri)
+        uri_list[idx : idx + 3] = ":__"
+        uri = "".join(uri_list)
+        uri = urlsplit(uri)
+        uri = [e for e in uri]
+        # replace back :__ with ://
+        uri[1] = uri[1].replace(":__", "://")
+        uri = SplitResult(uri[0], uri[1], uri[2], uri[3], uri[4])
```

Comment:
Why do you need to do this, `uri` is already a `SplitResult` instance…?

Reply:
Thanks for your comment. No, one line 200 we are doing a list comprehension. This is leading to the type being as `list`.

Verified by adding a few log statements:
```
def split_uri_to_parts(uri):
    c = uri.count("://")
    if c < 2:
        return urlsplit(uri), False
    idx = uri.rfind("://")
    # replace :// with :__
    uri_list = list(uri)
    uri_list[idx : idx + 3] = ":__"
    uri = "".join(uri_list)
    uri = urlsplit(uri)
    uri = [e for e in uri]
    # replace back :__ with ://
    uri[1] = uri[1].replace(":__", "://")

    print(type(uri))

    uri = SplitResult(uri[0], uri[1], uri[2], uri[3], uri[4])

    print(type(uri))
    return uri, True
```

Output is:
```
<class 'list'>
<class 'urllib.parse.SplitResult'>
(SplitResult(scheme='spark', netloc='k8s://100.68.0.1:443', path='', query='deploy-mode=cluster', fragment=''), True)
```

Reply:
And why do you need to do that list comprehention in the first place? SplitResult supports index access out of the box.

Reply:
Hello, I am trying to re-assign a variable to SplitResult list. That list does not support reassignment. Hence to achieve something like that, I had to convert it to a list, re-assign the elements and then convert it back to the SplitResult list. @uranusjr 
```
/Users/adesai/.pyenv/versions/airflow-env/bin/python /Users/adesai/Documents/OpenSource/Airflow/airflow/airflow/models/test.py
Traceback (most recent call last):
  File "/Users/adesai/Documents/OpenSource/Airflow/airflow/airflow/models/test.py", line 24, in <module>
    parts = split_uri_to_parts("spark://k8s://100.68.0.1:443?deploy-mode=cluster")
  File "/Users/adesai/Documents/OpenSource/Airflow/airflow/airflow/models/test.py", line 15, in split_uri_to_parts
    uri[1] = uri[1].replace(":__", "://")
TypeError: 'SplitResult' object does not support item assignment
```

Reply:
You should use `_replace`. See documentation on namedtuple.

Reply:
Thank you! Checked it out and it is brilliant. Pushing a commit with the fix.